### ISSUE TITLE
Scan multiple key values in one CF at once

### ISSUE THREAD

**Issue by breeswish** [tikv/tikv#4238]
## Feature Request

Currently in mvcc reader we step each cursor (write and lock cursor) one by one in order to get exactly data we want. However this might be not efficient, for example, cache might be not utilized well due to frequent cursor switch. 

We can try to scan multiple keys (and buffer it) at once to see if it improves performance. For example, we can first scan many keys in the lock cf. Then scan many keys in the write cf. For unused keys we keep it in the buffer for later use.

**Comment by siddontang** [comment]
I think we can at first seek to a lock key like a, then get the next lock key like b, then we can use this lock range [a, b) to scan in the write CF.

**Comment by AndreMouche** [comment]
For `read` in the coprocessor, it seems to get one lock should be enough since we will return when meeting the first lock.

**Comment by breeswish** [comment]
@AndreMouche Yes. But maybe our implementation need to take care of it to keep compatibility with KV interface (KV interface doesn't stop when there is a lock). Additionally we also need to batch the read of write CF and data CF.
